File: apps/admin-web/src/app/shared/rule.models.ts
// Decision types
export const DecisionTypes = {
  AutoPublish: 'AutoPublish',
  RequireApproval: 'RequireApproval',
  Block: 'Block',
  Schedule: 'Schedule'
} as const;

export type DecisionType = typeof DecisionTypes[keyof typeof DecisionTypes];

export const DecisionTypeOptions: { value: DecisionType; label: string }[] = [
  { value: 'AutoPublish', label: 'Auto Publish' },
  { value: 'RequireApproval', label: 'Require Approval' },
  { value: 'Block', label: 'Block' },
  { value: 'Schedule', label: 'Schedule' }
];

// Content status types
export const ContentStatuses = {
  New: 'New',
  PendingApproval: 'PendingApproval',
  Blocked: 'Blocked',
  Scheduled: 'Scheduled',
  AutoReady: 'AutoReady',
  Published: 'Published',
  Archived: 'Archived'
} as const;

export type ContentStatus = typeof ContentStatuses[keyof typeof ContentStatuses];

export const ContentStatusOptions: { value: ContentStatus; label: string }[] = [
  { value: 'New', label: 'New' },
  { value: 'PendingApproval', label: 'Pending Approval' },
  { value: 'Blocked', label: 'Blocked' },
  { value: 'Scheduled', label: 'Scheduled' },
  { value: 'AutoReady', label: 'Auto Ready' },
  { value: 'Published', label: 'Published' },
  { value: 'Archived', label: 'Archived' }
];

// Rule interfaces
export interface Rule {
  id: string;
  name: string;
  isEnabled: boolean;
  priority: number;
  decisionType: DecisionType;
  minTrustLevel: number | null;
  keywordsIncludeCsv: string | null;
  keywordsExcludeCsv: string | null;
  sourceIdsCsv: string | null;
  groupIdsCsv: string | null;
  createdAtUtc: string;
  createdByUserId: string | null;
  createdByUserEmail: string | null;
}

export interface CreateRuleRequest {
  name: string;
  isEnabled: boolean;
  priority: number;
  decisionType: DecisionType;
  minTrustLevel?: number | null;
  keywordsIncludeCsv?: string | null;
  keywordsExcludeCsv?: string | null;
  sourceIdsCsv?: string | null;
  groupIdsCsv?: string | null;
}

export interface UpdateRuleRequest {
  name: string;
  isEnabled: boolean;
  priority: number;
  decisionType: DecisionType;
  minTrustLevel?: number | null;
  keywordsIncludeCsv?: string | null;
  keywordsExcludeCsv?: string | null;
  sourceIdsCsv?: string | null;
  groupIdsCsv?: string | null;
}

export interface RecomputeRequest {
  fromUtc?: string;
  toUtc?: string;
  sourceId?: string;
  status?: string;
}

export interface RecomputeResult {
  processed: number;
  changed: number;
  byDecisionTypeCounts: { [key: string]: number };
}

export interface RuleDecisionResult {
  decisionType: DecisionType;
  status: ContentStatus;
  ruleId: string | null;
  ruleName: string | null;
  reason: string;
  scheduledAtUtc: string | null;
}

File: apps/api/HaberPlatform.Api/Models/RuleModels.cs
using System.ComponentModel.DataAnnotations;
using HaberPlatform.Api.Entities;

namespace HaberPlatform.Api.Models;

// DTO for Rule list/detail
public class RuleDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public bool IsEnabled { get; set; }
    public int Priority { get; set; }
    public string DecisionType { get; set; } = string.Empty;
    public int? MinTrustLevel { get; set; }
    public string? KeywordsIncludeCsv { get; set; }
    public string? KeywordsExcludeCsv { get; set; }
    public string? SourceIdsCsv { get; set; }
    public string? GroupIdsCsv { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public Guid? CreatedByUserId { get; set; }
    public string? CreatedByUserEmail { get; set; }
}

// Create request
public class CreateRuleRequest
{
    [Required]
    [StringLength(255, MinimumLength = 1)]
    public string Name { get; set; } = string.Empty;
    
    public bool IsEnabled { get; set; } = true;
    
    [Range(0, int.MaxValue)]
    public int Priority { get; set; } = 0;
    
    [Required]
    public string DecisionType { get; set; } = DecisionTypes.RequireApproval;
    
    public int? MinTrustLevel { get; set; }
    public string? KeywordsIncludeCsv { get; set; }
    public string? KeywordsExcludeCsv { get; set; }
    public string? SourceIdsCsv { get; set; }
    public string? GroupIdsCsv { get; set; }
}

// Update request
public class UpdateRuleRequest
{
    [Required]
    [StringLength(255, MinimumLength = 1)]
    public string Name { get; set; } = string.Empty;
    
    public bool IsEnabled { get; set; } = true;
    
    [Range(0, int.MaxValue)]
    public int Priority { get; set; } = 0;
    
    [Required]
    public string DecisionType { get; set; } = DecisionTypes.RequireApproval;
    
    public int? MinTrustLevel { get; set; }
    public string? KeywordsIncludeCsv { get; set; }
    public string? KeywordsExcludeCsv { get; set; }
    public string? SourceIdsCsv { get; set; }
    public string? GroupIdsCsv { get; set; }
}

// Rule decision result
public class RuleDecisionResult
{
    public string DecisionType { get; set; } = DecisionTypes.RequireApproval;
    public string Status { get; set; } = ContentStatuses.PendingApproval;
    public Guid? RuleId { get; set; }
    public string? RuleName { get; set; }
    public string Reason { get; set; } = "Default";
    public DateTime? ScheduledAtUtc { get; set; }
}

// Recompute request
public class RecomputeRequest
{
    public DateTime? FromUtc { get; set; }
    public DateTime? ToUtc { get; set; }
    public Guid? SourceId { get; set; }
    public string? Status { get; set; }
}

// Recompute result
public class RecomputeResult
{
    public int Processed { get; set; }
    public int Changed { get; set; }
    public Dictionary<string, int> ByDecisionTypeCounts { get; set; } = new();
}

// Extension methods for mapping
public static class RuleExtensions
{
    public static RuleDto ToDto(this Rule rule)
    {
        return new RuleDto
        {
            Id = rule.Id,
            Name = rule.Name,
            IsEnabled = rule.IsEnabled,
            Priority = rule.Priority,
            DecisionType = rule.DecisionType,
            MinTrustLevel = rule.MinTrustLevel,
            KeywordsIncludeCsv = rule.KeywordsIncludeCsv,
            KeywordsExcludeCsv = rule.KeywordsExcludeCsv,
            SourceIdsCsv = rule.SourceIdsCsv,
            GroupIdsCsv = rule.GroupIdsCsv,
            CreatedAtUtc = rule.CreatedAtUtc,
            CreatedByUserId = rule.CreatedByUserId,
            CreatedByUserEmail = rule.CreatedByUser?.Email
        };
    }
}

